data-team-uhn/heracles
# HERACLES-1: The Dockerfile for HERACLES should specify the CARDS_PROJECT and CARDS_PROJECT_VERSION environment variables

This _Pull Request_ fixes the environment variables set by default in the HERACLES Docker container to match with the environment variables expected by the `distribution/docker_entry.sh` script.

https://github.com/data-team-uhn/cards/blob/7afeb1eb32335bde369cae1dc27ef598765c9f0f/distribution/docker_entry.sh#L50-L53.

To test:

1. Build a self-contained `cards/cards:latest` Docker image from the `CARDS-2317` branch of the https://github.com/data-team-uhn/cards repository. You can do this by cloning https://github.com/data-team-uhn/cards, switching to the `CARDS-2317` branch, entering the `Utilities/Packaging/Docker` directory and running `./build_self_contained.sh cards/cards:latest`. You'll need to have a `cards/sling-feature-downloader` Docker image available locally (https://github.com/data-team-uhn/cards-sling-feature-downloader).
2. Clone the https://github.com/data-team-uhn/cards-parent repository, checkout the `cards-parent-1` tag, and run `mvn clean install`.
4. Clone this (https://github.com/data-team-uhn/heracles) repository and switch to this `HERACLES-1` branch.
5. Run `./get_cards_platform_jars.sh` and select a `local` image.
6. Run `mvn clean install -P production,latest`
7. Clone the https://github.com/data-team-uhn/cards-deploy-tool repository and run `python3 generate_compose_yaml.py --oak_filesystem --cards_docker_image cards/cards4heracles:latest --composum`
8. Edit the `docker-compose.yml` file and remove the `ADDITIONAL_SLING_FEATURES` line.
9. `docker-compose build`
10. `docker-compose up -d`
11. Ensure that HERACLES starts at http://localhost:8080
12. `docker-compose down && docker-compose rm && docker volume prune -f && ./cleanup.sh`
